backtrace-library/src/androidTest/java/backtraceio/library/BacktraceClientMetricsTest.java

```diff
@@ -0,0 +1,938 @@
+package backtraceio.library;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static junit.framework.TestCase.fail;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static java.lang.Thread.sleep;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import net.jodah.concurrentunit.Waiter;
+
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.common.DeviceAttributesHelper;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.logger.LogLevel;
+import backtraceio.library.metrics.BacktraceMetrics;
+import backtraceio.library.metrics.Event;
+import backtraceio.library.metrics.EventsOnServerResponseEventListener;
+import backtraceio.library.metrics.EventsPayload;
+import backtraceio.library.metrics.EventsRequestHandler;
+import backtraceio.library.metrics.EventsResult;
+import backtraceio.library.metrics.UniqueEvent;
+import backtraceio.library.models.BacktraceMetricsSettings;
+import backtraceio.library.models.json.BacktraceAttributes;
+import backtraceio.library.models.types.BacktraceResultStatus;
+
+@RunWith(AndroidJUnit4.class)
+public class BacktraceClientMetricsTest {
+    public Context context;
+    public BacktraceClient backtraceClient;
+    public BacktraceCredentials credentials;
+    private final String summedEventName = "activity-changed";
+    // existing attribute name in Backtrace
+    private final String[] uniqueAttributeName = {"uname.version", "cpu.boottime", "screen.orientation", "battery.state", "device.airplane_mode", "device.sdk", "device.brand", "system.memory.total", "uname.sysname", "application.package"};
+
+    private final String defaultBaseUrl = BacktraceMetrics.defaultBaseUrl;
+    private final String token = "aaaaabbbbbccccf82668682e69f59b38e0a853bed941e08e85f4bf5eb2c5458";
+    private final String universeName = "testing-universe-name";
+    private final String _uniqueEventsSubmissionUrl = "unique-events/submit?token=" + token + "&universe=" + universeName;
+    private final String _summedEventsSubmissionUrl = "summed-events/submit?token=" + token + "&universe=" + universeName;
+
+    static {
```

Comment:
Why do we need to load a native interface here?

Reply:
We probably don't need it, I will try to remove, thank you!